File: LAB QUIZZES/CSE221_Quiz05/_A6D/_A6D.py
import os, sys, time, random, threading, subprocess
from _A6D_Solution import solve
start = 0; finish = 0; elapsed = 0
test = 0; best = 0; score = 0; batch = 0; total = 0
timeLimit = 4; nBatch = 14; N = 0; R = 0; edges = []
weight = [ 0, 5, 5, 5, 5, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8]
nTest =  [0, 1, 1, 1, 1, 40, 40, 40, 40, 20, 20, 10, 10, 5, 5]
res = []

def printCase(verdict, test, N, R, edges):
    if os.path.getsize(verdict + ".txt") == 0:
        with open(verdict + ".txt", "w") as fout:
            fout.write("batch = " + str(batch) + "\ntest = " + str(test) + "\n")
            fout.write("N = " + str(N) + "\n")
            fout.write("R = " + str(R) + "\n")
            fout.write("edges = " + str(edges) + "\n")
    print(verdict + " on Batch " + str(batch), flush=True)
def exitCase(verdict):
    printCase(verdict, test, N, R, edges)
    os._exit(1)
def checkSoln():
    global start, finish, elapsed, N, R, edges
    start = time.time()
    res = solve(N, R, edges)
    finish = time.time()
    elapsed += finish - start
    if len(res) != N + 1:
        exitCase("WrongAnswer")
    with open("_A6D_output.txt", "w") as fout:
        res[0] = 0
        for i in range(0, N + 1):
            fout.write(str(res[i]) + " ")
        fout.write("\n")


    ret = subprocess.run('_A6D_Judge.out _A6D_input.txt _A6D_output.txt', shell = True)
    if ret.returncode != 0:
        exitCase("WrongAnswer")

def processBatch():
    global test, N, R, edges
    sys.setrecursionlimit(10**6)
    test = 1
    while test <= nTest[batch]:
        cmd = "_A6D_Gen.out " + str(batch) + " " + str(test) + " > _A6D_input.txt"
        subprocess.run(cmd, shell = True)
        with open('_A6D_input.txt') as temp:
            N, R = list(map(int, temp.readline().split()))
            edges = []
            for _ in range(N-1):
                u, v = list(map(int, temp.readline().split()))
                edges.append((u, v))
        
        threading.stack_size(128 * 1024 * 1024)     
        thread = threading.Thread(target=checkSoln, daemon=True)
        thread.start()
        thread.join()
        test += 1

def limitTime():
    recall = -1
    while elapsed < timeLimit and recall != elapsed:
        recall = elapsed
        time.sleep(timeLimit << 1)
    exitCase("TimeLimitExceeded")
if len(sys.argv) == 2:
    batch = int(sys.argv[1])
    random.seed(batch)
if 1 <= batch and batch <= nBatch:
    print("Running on Batch " + str(batch), flush=True)
    threading.Thread(target=limitTime, daemon=True).start()
    processBatch()
    print("Accepted on Batch " + str(batch) + " in " + f"{elapsed:.9f}" + "s", flush=True)
    os._exit(0)
ID = ""
with open("EnterIDandLanguage.txt") as fin:
    ID = fin.readline().split()[0]
if os.path.exists("_A6D_" + ID + ".py"):
    with open("_A6D_" + ID + ".py") as fin:
        best = int(fin.readline().split()[2])
open("WrongAnswer.txt", "w").close()
open("TimeLimitExceeded.txt", "w").close()
batch = 1
while batch <= nBatch:
    process = subprocess.run("pypy _A6D.py " + str(batch), capture_output=True, text=True)
    if process.stdout:
        print(process.stdout, end="")
    if process.stderr:
        print(process.stderr, end="")
    if process.returncode == 0:
        score += weight[batch]
    total += weight[batch]
    batch += 1
if best <= score:
    with open("_A6D_" + ID + ".py", "w") as fout:
        fout.write("# " + ID + " " + str(score) + "\n")
    subprocess.run("cmd /c echo # %COMPUTERNAME% %USERNAME%>>_A6D_" + ID + ".py")
    with open("_A6D_" + ID + ".py", "a") as fout:
        with open("_A6D_Solution.py") as code:
            fout.write(code.read())
print("Tentative score = " + str(score / total) + "/1", flush=True)
os._exit(0)

File: LAB QUIZZES/CSE221_Quiz05/_A6D/_A6D.java
import java.util.*;
import java.io.*;

public class _A6D {
    public static Random rng = new Random();
    public static long start, finish, elapsed = 0, MOD = 1000003;
    public static int test = 0, best = 0, score = 0, batch = 0, total = 0;
    public static int timeLimit = 2, nBatch = 14;
    public static int weight[] = { 0, 5, 5, 5, 5, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8};
     public static int nTest[] = { 0, 1, 1, 1, 1, 40, 40, 40, 40, 20, 20, 10, 10, 5, 5};
    public static int N, R;
    public static ArrayList<Integer> res;
    public static ArrayList<_A6D_Solution.Pair<Integer, Integer>> edges = new ArrayList<>();

    public static void printCase(String verdict, int test, int N, int R, ArrayList<_A6D_Solution.Pair<Integer, Integer>> edges)
            throws Exception {
        if (new File(verdict + ".txt").length() == 0) {
            BufferedWriter fout = new BufferedWriter(new FileWriter(verdict + ".txt"));
            fout.write("batch = " + batch + "\ntest = " + test + "\n");
            fout.write("N = " + N + "\n");
            fout.write("R = " + R + "\n");
            fout.write("edges = " + edges + "\n");
            fout.close();
        }
        System.out.print(verdict + " on Batch " + batch + "\n");
    }

    public static void exitCase(String verdict) throws Exception {
        printCase(verdict, test, N, R, edges);
        System.exit(1);
    }

    public static void checkSoln() throws Exception {
        start = System.nanoTime();
        res = _A6D_Solution.solve(N, R, edges);
        finish = System.nanoTime();
        elapsed += finish - start;
        if (1000000000L * timeLimit < elapsed)
            exitCase("TimeLimitExceeded");
        if (res.size() != N + 1)
            exitCase("WrongAnswer");

        BufferedWriter fout = new BufferedWriter(new FileWriter("_A6D_output.txt"));
        res.set(0, 0);
        for (int i = 0; i <= N; ++i) {
            fout.write(res.get(i) + " ");
        }
        fout.write("\n");
        fout.close();

        Process process = new ProcessBuilder("cmd.exe", "/c", "_A6D_Judge.out  _A6D_input.txt _A6D_output.txt").start();
        process.waitFor();
        if (process.exitValue() != 0)
            exitCase("WrongAnswer");
    }

    public static void processBatch() throws Exception {
        
        for (test = 1; test <= nTest[batch]; ++test) {
            String cmd = "_A6D_Gen.out " + batch + " " + test + " > _A6D_input.txt";
            Process process = new ProcessBuilder("cmd.exe", "/c", cmd).start();
            process.waitFor();

            BufferedReader fin = new BufferedReader(new FileReader("_A6D_input.txt"));
            String[] nums = fin.readLine().split(" ");
            N = Integer.parseInt(nums[0]);
            R = Integer.parseInt(nums[1]);
            edges.clear();
            for (int i = 0; i < N - 1; i++) {
                String[] temp = fin.readLine().split(" ");
                int u = Integer.parseInt(temp[0]);
                int v = Integer.parseInt(temp[1]);
                edges.add(new _A6D_Solution.Pair<>(u, v));
            }
            checkSoln();
        }
    }

    public static void limitTime() {
        try {
            for (long recall = -1; elapsed < 1000000000L * timeLimit && recall != elapsed;) {
                recall = elapsed;
                Thread.sleep(1000 * timeLimit << 1);
            }
            exitCase("TimeLimitExceeded");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void main(String args[]) throws Exception {
        if (args.length == 1)
            rng.setSeed(batch = Integer.parseInt(args[0]));
        if (1 <= batch && batch <= nBatch) {
            System.out.println("Running on Batch " + batch);
            new Thread(() -> limitTime()).start();
            processBatch();
            System.out.printf("Accepted on Batch " + batch + " in %.9fs\n", elapsed * 1e-9);
            System.exit(0);
        }
        BufferedReader fin = new BufferedReader(new FileReader("EnterIDandLanguage.txt"));
        String ID = fin.readLine().split("\\s+", 2)[0];
        fin.close();
        if (new File("_A6D_" + ID + ".java").exists()) {
            fin = new BufferedReader(new FileReader("_A6D_" + ID + ".java"));
            best = Integer.parseInt(fin.readLine().split("\\s+", 4)[2]);
            fin.close();
        }
        new FileOutputStream("WrongAnswer.txt").close();
        new FileOutputStream("TimeLimitExceeded.txt").close();
        for (batch = 1; batch <= nBatch; total += weight[batch], ++batch) {
            Process process = new ProcessBuilder("java", "-Xss128m", "_A6D", "" + batch).start();
            fin = new BufferedReader(new InputStreamReader(process.getInputStream()));
            fin.lines().forEach(System.out::println);
            fin.close();
            fin = new BufferedReader(new InputStreamReader(process.getErrorStream()));
            fin.lines().forEach(System.out::println);
            fin.close();
            if (process.waitFor() == 0)
                score += weight[batch];
        }
        if (best <= score) {
            BufferedWriter fout = new BufferedWriter(new FileWriter("_A6D_" + ID + ".java"));
            fout.write("// " + ID + " " + score + "\n");
            fout.close();
            new ProcessBuilder("cmd", "/c", "echo // %COMPUTERNAME% %USERNAME%>>_A6D_" + ID + ".java").start()
                    .waitFor();
            fout = new BufferedWriter(new FileWriter("_A6D_" + ID + ".java", true));
            fin = new BufferedReader(new FileReader("_A6D_Solution.java"));
            fout.write(fin.lines().collect(java.util.stream.Collectors.joining(System.lineSeparator())));
            fin.close();
            fout.close();
        }
        System.out.print("Tentative score = " + (score / (double) total) + "/1\n");
        System.exit(0);
    }
}
